### PR TITLE
It lives again (Asteroid-station)

### PR DESCRIPTION
**Pull request by CorruptComputer**
One of the main complaints I heard was maint having too many 1 wide hallways, and being too boring. I have fixed this, however if anyone had other issues feel free to comment.

The file was too large to attach, but here's the link: https://drive.google.com/file/d/0B1-dAHGygVfZOEtrMHF2QVNUVTA/view?usp=sharing

### CODE REVIEW

**Comment by as334** [comment]
Ass is back boys.

**Comment by KorPhaeron** [comment]
Asteroids are no longer canon, sorry

**Comment by KazeEspada** [comment]
Needs lavaland.

**Comment by Zilenan91** [comment]
Someone should make a lavaland station

**Comment by CorruptComputer** [comment]
@KazeEspada AsteroidStation doesn't even have a mining station as it is included in the main station itself, what would lavaland add?

**Comment by ChangelingRain** [comment]
Can you even OPEN this? I'm pretty sure a map this large crashes byond if you try to compile it with tgm.

**Comment by CorruptComputer** [comment]
@ChangelingRain I fixed that issue as well

**Comment by CorruptComputer** [comment]
@KazeEspada Actually after thinking about it, that would be kind of cool, I could set it up so the shuttle says "Takes passengers deep within the asteroid" or something and it could be RP friendly.

**Comment by ChangelingRain** [comment]
Fixed... how?

**Comment by CorruptComputer** [comment]
I reduced the file size from 340x340 down to 255x255, making it the same as all other maps. To do this I did need to slim down the asteroid though.

**Comment by ChangelingRain** [comment]
That works, I guess!

File: _maps/asteroidstation.dm [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+		#define MINETYPE "mining"
+
+        #include "map_files\AsteroidStation\AsteroidStation.dmm"
+        #include "map_files\generic\z2.dmm"
+        #include "map_files\generic\z3.dmm"
+        #include "map_files\generic\z4.dmm"
+        #include "map_files\AsteroidStation\z5.dmm"
+        #include "map_files\generic\z6.dmm"
+        #include "map_files\generic\z7.dmm"
```
Comment (Incoming5643) [inline _maps/asteroidstation.dm]: we got z8 and z9 now too

File: _maps/asteroidstation.dm [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+		#define MAP_PATH "map_files/AsteroidStation"
+        #define MAP_FILE "AsteroidStation.dmm"
+        #define MAP_NAME "AsteroidStation"
+
+        #define MAP_TRANSITION_CONFIG	list(MAIN_STATION = CROSSLINKED, CENTCOMM = SELFLOOPING, ABANDONED_SATELLITE = CROSSLINKED, DERELICT = CROSSLINKED, MINING = CROSSLINKED, EMPTY_AREA_1 = CROSSLINKED, EMPTY_AREA_2 = CROSSLINKED)
```
Comment (Incoming5643) [inline _maps/asteroidstation.dm]: add `EMPTY_AREA_3 = CROSSLINKED, EMPTY_AREA_4 = CROSSLINKED`

**Comment by KazeEspada** [comment]
Elevator to lavaland.

File: _maps/asteroidstation.dm (hunk 1 above)
Comment (CorruptComputer) [inline _maps/asteroidstation.dm]: Jesus, when did these happen, also the map works fine without them. But I will add tomorrow, for now DNM I guess.

File: _maps/asteroidstation.dm (hunk 2 above)
Comment (CorruptComputer) [inline _maps/asteroidstation.dm]: Same as above.

**Comment by Core0verload** [comment]
Why do we need all those maps when the only one that matters is Box?

**Comment by CorruptComputer** [comment]
@Core0verload What's the point of box when we have all these other maps?

**Comment by Core0verload** [comment]
@CorruptComputer, to play on it, obviously.

It's simple. Box is the map you play on. Other ones are the maps that cause Travis to raise red flag on your commits.

**Comment by coiax** [comment]
So is this an AsteroidStation on OLD ASTEROID, or on NEW LAVALAND? Because if we can stick it on lavaland, that'd be mighty sweet. No idea if that's possible though, does Lavaland have enough space?

**Comment by CorruptComputer** [comment]
@coiax Uh...? I'm not quite sure what you're asking, it's just an updated version of AsteroidStation.

**Comment by Dorsisdwarf** [comment]
On basil Metastation is the one you play on, with occasional holidays to Box or Dream (everyone hates dream)

**Comment by KazeEspada** [comment]
Make the mining shuttle into an elevator with chasms as the base turf. So you can drop down onto lavaland, IE lavaland is really underneath asteroid station.

**Comment by CorruptComputer** [comment]
Well currently there is no mining shuttle, but I'll see what I can do.

**Comment by coiax** [comment]
Yeah, either use a mining shuttle or bring back ladders.

**Comment by CorruptComputer** [comment]
@coiax So after looking into this, I can't quite get it working the way I would like, so I am going to leave that for a later PR.

File: _maps/asteroidstation.dm [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+        #include "map_files\AsteroidStation\AsteroidStation.dmm"
+        #include "map_files\generic\z2.dmm"
+        #include "map_files\generic\z3.dmm"
+        #include "map_files\generic\z4.dmm"
+        #include "map_files\AsteroidStation\z5.dmm"
+        #include "map_files\generic\z6.dmm"
+        #include "map_files\generic\z7.dmm"
+        #include "map_files\generic\z8.dmm"
+		#include "map_files\generic\z9.dmm"
```
Comment (KorPhaeron) [inline _maps/asteroidstation.dm]: Is this the online viewer fucking up or is it indented a bunch

Comment (CorruptComputer) [inline _maps/asteroidstation.dm]: It's the online viewer, they look fine in NP++ for me.

Comment (ChangelingRain) [inline _maps/asteroidstation.dm]: All the rest are indented with spaces, this is indented with two tabs.

Comment (CorruptComputer) [inline _maps/asteroidstation.dm]: Oh, didn't even notice, will fix later today.

**Comment by Iamgoofball** [comment]
i fucking KNEW it was due to assteroid being too big we couldn't launch it with tgm

**Comment by Razharas** [comment]
Is this ready @CorruptComputer ?

**Comment by CorruptComputer** [comment]
Yeah it's good.

**Comment by Razharas** [comment]
I really hope you cant make black holes for free or craft anything from anything here

**Pull request merged**